sdk/dynamicconfig/metric/controller/push/push.go

```diff
@@ -172,14 +172,13 @@ func (c *Controller) tick() {
 }
 
 func (c *Controller) update(schedules []*pb.MetricConfigResponse_Schedule) {
-	c.matcher.ConsumeSchedules(schedules)
-	minPeriod := c.matcher.GetMinPeriod()
-	if c.lastPeriod != minPeriod {
+	newPeriod := c.matcher.ApplySchedules(schedules)
+	if c.lastPeriod != newPeriod {
 		if c.ticker != nil {
 			c.ticker.Stop()
 		}
 
-		c.lastPeriod = minPeriod
+		c.lastPeriod = newPeriod
```

Comment:
Let's change lastPeriod to exportPeriod? The "last" part does not carry too much information and is only useful in "update"; for the rest of the code, lastPeriod is actually current exportPeriod. The benefit of "exportPeriod" is that it is differentiated from any period from the schedule.

Reply:
Good idea, changing